### PR TITLE
default proxy mode feature

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

**Review by Aceix — APPROVED** [review]

File: functions/join.go [hunk 1]
```diff
@@ -335,6 +335,7 @@ func JoinNetwork(flags *viper.Viper) (*config.Node, *config.Server, error) {
 	}
 	logger.Log(1, "network:", node.Network, "node created on remote server...updating configs")
 	config.UpdateServerConfig(&joinResponse.ServerConfig)
+	config.UpdateHost(&joinResponse.Host)
```
Comment (0xdcarns) [inline functions/join.go:338]: will also need to call this in register

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Comment by afeiszli** [comment]
How is default proxy mode being set here?

File: functions/join.go (hunk 1 above)
Comment (mattkasun) [reply functions/join.go:338]: registerResponse does not include host data so this call cannot be made from register function

**Review by 0xdcarns — APPROVED** [review]

File: config/config.go [hunk 2]
```diff
@@ -75,6 +75,29 @@ func UpdateNetclient(c Config) {
...
+		netclient.Host.ListenPort = newHost.ListenPort
+	}
+	if newHost.ProxyListenPort > 0 {
+		netclient.Host.ProxyListenPort = newHost.ProxyListenPort
+	}
+	netclient.Host.IsDefault = newHost.IsDefault
+	netclient.Host.DefaultInterface = newHost.DefaultInterface
+	if !netclient.Host.ProxyEnabledSet {
+		netclient.Host.ProxyEnabled = newHost.ProxyEnabled
+		netclient.Host.ProxyEnabledSet = true
```
Comment (abhishek9686) [inline config/config.go:93]: this if block `if !netclient.Host.ProxyEnabledSet` is not clear to me, what is the intention of this?

**Comment by theguy951357** [comment]
will wait on changes

File: config/config.go (hunk 2 above)
Comment (mattkasun) [reply config/config.go:93]: This is to prevent a second server from changing the value it it was set by a previous server

**Review by theguy951357 — APPROVED** [review]
testing steps in 1375 passes
critical checklist passes

**Pull request merged**